### PR TITLE
Updating readme for swagger & run errors. clean-up & update swagger.

### LINKED ISSUE #42: How to regenerate SWAGGER-based code without needing to delete prev code each time
Find the way SWAGGER recommends we do this, document the process in README.

### LINKED ISSUE #71: Figure out why using the latest code generator doesn't seem to work
We had to pin to an older version in https://github.com/sul-dlss-labs/taco/pull/70.  Figure out why the newer version is different.

### LINKED ISSUE #86: Swagger json should have response samples for each endpoint

### LINKED ISSUE #87: Swagger request for file deposit looks incorrect
It's showing the body to be JSON, but shouldn't it be the binary itself?

Shouldn't it accept a request such as:
```
curl -X POST -d @README.md http://localhost:8080/file
```

### LINKED ISSUE #101: Define the GET /status route in swagger.json

### LINKED ISSUE #103: Add some required fields to the request.
Add required fields:

publish, access, preserve, label, `@type`, `@context` as required.

### LINKED ISSUE #105: POST /resource should provide an error if they try to provide an ID.

### LINKED ISSUE #110: update local run command (incorrect) on README

### LINKED ISSUE #111: update README.md - dont need go get -t for local binary build

### PR DESCRIPTION
**Pull request by cmh2166**
README errors:

- [x] closes #42. 
- [x] closes #71. 
- [x] closes #110. 
- [x] closes #111

Swagger repairs:
- [x] rewrite Swagger per latest discussion
- [x] refs #105 
- [x] closes #101 
- [x] closes #103 
- [x] closes #86 
- [x] closes #87 and opens #149 (question around file upload handling)

### CODE REVIEW

**Comment by jcoyne** [comment]
Is there some reason this README change can’t stand on its own and be considered complete as it is now?

File: swagger.json [hunk 1]
```diff
@@ -175,6 +175,22 @@
           "type" : "string",
           "description": "What groups should be able to access the resource in Access environments"
         },
+        "contains" : {
```
Comment (jcoyne) [inline swagger.json]: Is there a reason we need to support both contains and containedBy?  Having two relationships that can be set greatly increases complexity in order to make consistency guarantees.

File: swagger.json [hunk 2]
```diff
@@ -224,21 +240,79 @@
...
+        },
+        "download" : {
           "type" : "string",
-          "example" : "mybook.pdf"
+          "description": "What groups should be able to download the File in Access environments"
+        },
+        "id" : {
+          "type" : "string",
+          "description": "The TACO identifier for the resource. Parent DRUID-derived.",
+          "example" : "oo000oo0001"
```
Comment (jcoyne) [inline swagger.json]: Should this be consistent with line 309?

File: swagger.json [hunk 3]
```diff
@@ -88,95 +97,274 @@
...
+        },
+        "contains" : {
+          "type" : "array",
+          "items" : {
+            "type" : "string",
+            "format" : "uri"
+          },
+          "description": "The child resource(s) of this resource."
+        },
+        "contained-by" : {
```
Comment (jcoyne) [inline swagger.json]: Do we need to support both relationships?  Supporting both simultaneously can make it more difficult to achieve consistency guarantees.

File: swagger.json [hunk 4]
```diff
@@ -15,49 +15,52 @@
...
+          "401" : {
+            "description" : "You are not authorized to create a resource in TACO."
+          },
+          "415" : {
+            "description" : "Unsupported resource type provided. TACO resources should be handed over as JSON or JSON-LD."
+          },
+          "422" : {
+            "description" : "The resource JSON provided had an unspecified or unsupported field, or is otherwise unprocessable by TACO."
+          },
+          "500" : {
```
Comment (jcoyne) [inline swagger.json]: Can the 4xx/5xx responses have a schema so we can give the user helpful error messages?

File: swagger.json [hunk 5]
```diff
@@ -88,95 +97,274 @@
...
         "operationId" : "depositNewFile",
         "produces" : [ "application/json" ],
         "parameters" : [ {
           "in" : "body",
           "name" : "body",
-          "description" : "File / Binary.",
+          "description" : "Binary to be added to an Object in TACO.",
           "required" : true,
           "schema" : {
             "$ref" : "#/definitions/File"
```
Comment (jcoyne) [inline swagger.json]: Was this what we talked about today?  Shouldn't this be a binary object?

Comment (cmh2166) [reply swagger.json]: it is.

File: swagger.json (hunk 4 above)
Comment (cmh2166) [reply swagger.json]: we haven't gotten there yet. we need to talk exception handling per today's decisions.

File: swagger.json (hunk 3 above)
Comment (cmh2166) [reply swagger.json]: yes.

File: swagger.json (hunk 5 above)
Comment (jcoyne) [reply swagger.json]: Doesn’t this point to 293 which indicates it’s a JSON object?

**Comment by jcoyne** [comment]
I would say this does not close #105 until we actually return errors in the handler and add a test to show that.

File: swagger.json (hunk 2 above)
Comment (aaron-collier) [reply swagger.json]: Can you elaborate? I'm not sure I understand the concern between this and 309.

File: swagger.json (hunk 3 above)
Comment (aaron-collier) [reply swagger.json]: I understand the concern @jcoyne, but if we're explicit about these relationships now, we should be able to stay on top of any difficulty/confusion.

**Comment by jcoyne** [comment]
@cmh2166 would you mind amending the description to say it closes #87 too?

**Pull request merged**